spec/models/change_service_event_spec.rb

```diff
@@ -21,6 +21,22 @@
       expect(change_service_event.data['name']).to eql(new_service_name)
     end
   end
+  context '#trigger_publish_event' do
+    it 'when change service name publish is trigger' do
+      new_entity_id = 'https://changed-entity-id'
+      service.assign_attributes(entity_id: new_entity_id)
+      change_service_event = create(:change_service_event, service: service)
+      component_1 = create(:sp_component)
+      event = create(:assign_sp_component_to_service_event, service: service, sp_component_id: component_1.id)
+      update_entity_id = 'https://changed-entity-id1'
+      service.assign_attributes(entity_id: update_entity_id)
```

Comment:
all of this could be replaced with just
```
assign_event = create(:assign_sp_component_to_service_event) # To make it valid/publishable component
updated_service = create(:service, entity_id: "http://changed-entity-id")
change_service_event = create(:change_service_event, service: updated_service)
```